### ISSUE TITLE
Fetch integration

### ISSUE THREAD

**Issue by annevk** [WICG/private-network-access#33]
Looking at what is labeled "ISSUE 3" in the specification it does seem reasonable to assume that once a connection is established there is IP address information, but currently that happens in https://fetch.spec.whatwg.org/#concept-http-network-fetch whereas some of the IP address checks in your algorithms seem to be placed elsewhere.

**Comment by letitz** [comment]
Woops, this should not be closed - there is still an ISSUE in the spec text referring to this.

Specifically, we want to use the notion of a [connection](https://fetch.spec.whatwg.org/#concept-connection)'s **remote endpoint** / **remote IP address**, and that does not yet exist in the spec. Maybe it is as simple as defining it as a new associated property of connections?

**Comment by annevk** [comment]
I'm looking into this as part of https://github.com/whatwg/fetch/issues/1243.

What's the set of steps here? Once we've obtained a connection we'll do a preflight if the connection's IP address fits the bill? Should "obtain a connection" end up calling into this spec?

**Comment by letitz** [comment]
The set of steps is now laid out in the [Fetch integration section](https://wicg.github.io/private-network-access/#secure-context-restriction). See the "Private Network Access check" algorithm and the modification to "HTTP-network fetch" below.

We do not need "obtain a connection" to call back into this spec. Instead, we perform the check on the connection once it is obtained, by looking at the IP address of its "remote endpoint", which is as yet unspecified.

If the Fetch spec ends up detailing how to resolve a domain to IP addresses, then obtaining a connection for that IP address, then we can reasonably define a property on the connection that stores the IP address, which would satisfy our needs here.

**Comment by annevk** [comment]
I guess it depends on what we want for WebTransport and WebRTC. I think the discussed setup is otherwise equivalent.

**Comment by letitz** [comment]
This because WebTransport and WebRTC rely on "obtain a connection" but not "HTTP-network fetch"?

**Comment by annevk** [comment]
Correct, although now you mention it WebTransport does some kind of CONNECT request that perhaps ought to use HTTP-network fetch.

**Comment by letitz** [comment]
Alright. I think I'll tackle the question of WebTransport and WebRTC separately from this issue right now.

It seems likely that we will soon have the IP address information in Fetch that would allow us to eliminate the handwaving in this spec. I'll just wait for that, then?

**Comment by annevk** [comment]
That seems reasonable. @ricea and I will try to make progress on it.
